src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -36,11 +36,26 @@ module type Inputs_intf = sig
      and type blockchain_state := Blockchain_state.value
      and type consensus_state := Consensus_state.value
 
+  (* TODO: This should replace the current existing external transition *)
```

Comment:
You should just do that now as part of this PR. I think the only major change is the extra arg to create, and only the proposer calls create afaik.